addons/account_edi/models/account_edi.py

```diff
@@ -0,0 +1,188 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields, api
+from odoo.exceptions import UserError
+from odoo.tools.pdf import OdooPdfFileReader
+
+from lxml import etree
+import base64
+import io
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountEdi(models.Model):
+    _name = 'account.edi'
+    _description = 'EDI format'
+
+    name = fields.Char(index=True)
+    code = fields.Char()
+    embed_to_pdf = fields.Boolean(default=True)
```

Comment:
Since every edi has its own implementation, maybe this field could be a method directly
```
def _embed_invoice_attachment_inside_pdf(self):
  self.ensure_one()
  return True
```

Reply:
This is not the method that check if the edi has been posted (to government for example), this is just a field stating if the file should be embed to pdf or sent has an attachment on export.

Previously, it was returned by _get_xml_decoders but since we created the model, it made more sense to put this kind of information as a field (_get_xml_decoders no longer exists).

The hook to check if an edi has been posted will be added later since it's not used in this task.

Reply:
something like `export_invoice = fields.Selection(selection=[('embed_to_pdf', '...'), ('separated_attachment', '...')])` to bring more flexibility maybe?